docs/source/autoscaling.rst

```diff
@@ -0,0 +1,34 @@
+====================================
+Autoscaling Marathon tasks in PaaSTA
+====================================
+
+PaaSTA allows programmatic control of the number of marathon tasks a service has. It does this by using Zookeeper as a data store to record the number of tasks it thinks should be running for each instance of a service on a per-cluster basis.
+
+How to use autoscaling
+======================
+
+Enabling autoscaling
+--------------------
+
+In order to use autoscaling, edit your ``marathon-*.yaml`` files in your soa configs and add a ``max_instances`` attribute to each instance you want to autoscale. When using autoscaling, the ``instances`` and ``max_instances`` attributes become the minimum and maximum (inclusive) number of marathon tasks paasta will create for your job.
+
+Adding a ``autoscaling_method`` attribute will allow you to specify the logic that determines when to scale up and scale down your instance. If an autoscaling method isn't provided, the ``"default"`` autoscaling method will be used.
```

Comment:
I think long term this might be better under a hash, like monitoring.
```
---
main:
  min_instances: 1
  max_instances: 5
  autoscaling:
    method: bespoke
```

My reasoning is that there is going to end up being a number of tunables and you will end up having to prefix them all with `autoscaling_`